DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -146,54 +361,949 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 #include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "TLorentzVector.h"
 void BPHMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
 
-//  edm::Handle<reco::BeamSpot> const& beamSpot;
-  // Filter out events if Trigger Filtering is requested
-  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
-
-  
   edm::Handle<reco::BeamSpot> beamSpot;
   iEvent.getByToken( bsToken_,  beamSpot);
+
   edm::Handle<reco::MuonCollection> muoHandle;
   iEvent.getByToken( muoToken_, muoHandle );
 
-  for (auto const & m : *muoHandle) {
-    if(!muoSelection_ref(m))continue;
-    muPhi_.denominator->Fill(m.phi());
-    muEta_.denominator->Fill(m.eta());
-    muPt_.denominator ->Fill(m.pt());
-    const reco::Track * track = 0;
-    if (m.isTrackerMuon()) track = & * m.innerTrack();
-    else if (m.isStandAloneMuon()) track = & * m.outerTrack();
-    if (track) {
-      double d0 = track->dxy(beamSpot->position());
-      double z0 = track->dz(beamSpot->position());
-      mud0_.denominator ->Fill(d0);
-      muz0_.denominator ->Fill(z0);
+  edm::Handle<reco::TrackCollection> trHandle;
+  iEvent.getByToken( trToken_, trHandle );
+
+  edm::Handle<reco::PhotonCollection> phHandle;
+  iEvent.getByToken( phToken_, phHandle );
+
+  edm::Handle<trigger::TriggerEvent> handleTriggerEvent; 
+//  edm::Handle<trigger::TriggerEventWithRefs> handleTriggerEvent; 
+//  iEvent.getByToken( hltInputTag_, handleTriggerEvent);
+//  if (handleTriggerEvent->sizeFilters()== 0)return;
+//  if(! handleTriggerEvent->isValid() ) return;
+//  edm::Handle<reco::BeamSpot> const& beamSpot;
+  // Filter out events if Trigger Filtering is requested
+  if (tnp_>0) {//TnP method 
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
```

Comment:
hi @fwyzard - i'll add my comments on this set of commits here as not to create a bunch of noise in your merged PR...